### DISCUSSION TITLE
Permissions and role based access for anonymous and user accounts

### DISCUSSION THREAD

**Discussion by churst90** [BearWare/TeamTalk5#2496] [Ideas]
Hello. I'd like to suggest a couple enhancements to permissions and account access to channels:
1. Cascading permissions for nested channels.
  Currently, my server has a public section and members section. Inside the channel root is a members category, inside which are 2 member channels. The same goes for the public category which contains 2 public channels. Currently, setting a password on the parent channel category does not cascade down to the channels inside it. This has caused me to have to set a password on each individual channel. However, this limits members from moving within this category since they'd need to know the password for each channel, leading to potential leaks, defeating the purpose of having account based access. Currently, if an account defaults to log into a members only channel, they can leave and rejoin the channel without entering a password, but only for the channel to which their account defaults to. This causes me as the admin to have to manually move people if they request it which is more overhead than I'd like. I feel this is different from channel operators as I don't necessarily want to grant them full permission to a channel if I don't want to. Maybe the ability to specify if a parent channel is a channel or a category, which might disallow people from attempting to join the parent channel. Currently I have to set the number of users for these parent channels to 0, but ideally it would be nice to just specify these channels as categories. Additionally, these categories should be able to have their own permissions, wherein an admin can add a user role, like 'members' to this category instead of having to manually set each account to view the category..
2. connections per account limitations:
  The server properties allows you to set how many connection maximum a user can have. The issue with this is that the public account on my channel has to abide by this rule which is not ideal. Under the account permissions for the anonymous account, there is a setting for disabling multiple logins. This is also not ideal. The number of connections needs to be able to be set on a per account basis. Some folks use their computers and/or phones to connect several times, but I would prefer not to afford this luxury to the anonymous account. I need folks to log in using the anonymous account but only once per IP. I currently cannot do this since it is either one connection period or multiple connections as set in the server properties.
3. User/channel over flow:
  If an account, such as a public anonymous account logs into the server, but the maximum number of users in that channel has been reached, they will get an error. However, it would be nice for them to still connect, but be placed in a second channel, such as a secondary public channel if the first one is full, rather than just disallowing them altogether.

**Comment by BlindMaster24** [comment]
> Hello. I'd like to suggest a couple enhancements to permissions and account access to channels:
> 
> 1. Cascading permissions for nested channels.
>    Currently, my server has a public section and members section. Inside the channel root is a members category, inside which are 2 member channels. The same goes for the public category which contains 2 public channels. Currently, setting a password on the parent channel category does not cascade down to the channels inside it. This has caused me to have to set a password on each individual channel. However, this limits members from moving within this category since they'd need to know the password for each channel, leading to potential leaks, defeating the purpose of having account based access. Currently, if an account defaults to log into a members only channel, they can leave and rejoin the channel without entering a password, but only for the channel to which their account defaults to. This causes me as the admin to have to manually move people if they request it which is more overhead than I'd like. I feel this is different from channel operators as I don't necessarily want to grant them full permission to a channel if I don't want to. Maybe the ability to specify if a parent channel is a channel or a category, which might disallow people from attempting to join the parent channel. Currently I have to set the number of users for these parent channels to 0, but ideally it would be nice to just specify these channels as categories. Additionally, these categories should be able to have their own permissions, wherein an admin can add a user role, like 'members' to this category instead of having to manually set each account to view the category..
> 2. connections per account limitations:
>    The server properties allows you to set how many connection maximum a user can have. The issue with this is that the public account on my channel has to abide by this rule which is not ideal. Under the account permissions for the anonymous account, there is a setting for disabling multiple logins. This is also not ideal. The number of connections needs to be able to be set on a per account basis. Some folks use their computers and/or phones to connect several times, but I would prefer not to afford this luxury to the anonymous account. I need folks to log in using the anonymous account but only once per IP. I currently cannot do this since it is either one connection period or multiple connections as set in the server properties.
> 3. User/channel over flow:
>    If an account, such as a public anonymous account logs into the server, but the maximum number of users in that channel has been reached, they will get an error. However, it would be nice for them to still connect, but be placed in a second channel, such as a secondary public channel if the first one is full, rather than just disallowing them altogether.

+
